bin/create-release

```diff
@@ -0,0 +1,200 @@
+
+with open('gh-release-token.txt', 'r') as file:
+    GH_TOKEN = file.read().rstrip()
+
+if not GH_TOKEN.startswith('ghp_'):
+    print('Invalid token found in gh-release-token.txt', file=sys.stderr)
+    exit(1)
+
+check_gh_auth_response = requests.get(
+    'https://api.github.com/repos/seattle-uat/civiform',
+    headers={'Authorization': f'token {GH_TOKEN}'})
+if check_gh_auth_response.status_code != 200:
+    print(
+        f'GitHub API auth failed, satus code: {str(check_gh_auth_response.status_code)}',
+        file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Check docker hub auth
+################################################################################
+
+if not 'Login Succeeded' in shell_cmd(
+        'docker login') or not 'civiform' in shell_cmd('docker info'):
+    print('Unable to login to docker hub')
+    exit(1)
+
+################################################################################
+# Validate release version number
+################################################################################
+
+if not re.match('^v\d+\.\d+\.\d+$', RELEASE_NUMBER):
+    print('Invalid version number: ' + RELEASE_NUMBER, file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Validate commit SHA
+################################################################################
+
+branches_containing_commit_sha = shell_cmd(
+    'git --no-pager branch --contains ' + COMMIT_SHA)
+branches_containing_commit_sha = [
+    # The output of the command is a list of branch names, with a '*' at the
+    # beginning of the current branch. This regex and the strip transform
+    # the list to only branch names with no trailing or preceding whitespace.
+    re.sub('\*', '', x).strip()
+    for x in branches_containing_commit_sha.split('\n')
+]
+if not 'main' in branches_containing_commit_sha:
+    print('Invalid commit SHA: ' + COMMIT_SHA, file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Find snapshot tag that matches commit SHA
+################################################################################
+
+# Git defaults to using the first 7 chars of the full SHA for the short SHA.
+SHORT_SHA = shell_cmd('git rev-parse --short ' + COMMIT_SHA)[:7]
+
+print(f'Searching for an image tag with short SHA {SHORT_SHA}')
+
+tags_url = 'https://registry.hub.docker.com/v2/repositories/civiform/civiform/tags?order=-last_activity&page_size=100'
+while True:
+    snapshot_tag = None
+    response = requests.get(tags_url)
+
+    if response.status_code != 200:
+        print(
+            f'Docker Hub request failed with status code: {str(response.status_code)}',
+            file=sys.stderr)
+        exit(2)
+
+    response_json = response.json()
+
+    for tag_info in response_json['results']:
```

Comment:
Is this relying on the github action that publishes images to tag them with the commit sha?  If so, can we document the dependancies? (Maybe an "if change" type block)